src/search3_cleanup.erl

```diff
@@ -0,0 +1,45 @@
+-module(search3_cleanup).
+
+-export([
+    clear_deleted_indexes/1
+]).
+
+-include("search3.hrl").
+-include_lib("couch/include/couch_db.hrl").
+-include_lib("fabric/include/fabric2.hrl").
+
+clear_deleted_indexes(Db) ->
+    #{
+        db_prefix := DbPrefix
+    } = Db,
+    fabric2_fdb:transactional(Db, fun(TxDb) ->
+        #{
+            tx := Tx
+        } = TxDb,
+        Indexes = search3_util:list_indexes(TxDb),
+        ExistingSigs  = [element(10, Index) || Index <- Indexes],
+        SigsFromList = get_sigs_from_list(Tx, DbPrefix),
+        ClearList = lists:filter(fun (Sig) ->
+            not lists:member(Sig, ExistingSigs)
+        end, SigsFromList),
+        clear_index_range(Tx, ClearList, DbPrefix),
+        remove_indexes_from_list(Tx, ClearList, DbPrefix)
+    end).
+
+get_sigs_from_list(Tx, DbPrefix) ->
+    Prefix = erlfdb_tuple:pack({?DB_SEARCH, ?INDEX_LIST}, DbPrefix),
+    Opts = [{streaming_mode, want_all}],
+    Rows = erlfdb:wait(erlfdb:get_range_startswith(Tx, Prefix, Opts)),
+    [element(6, erlfdb_tuple:unpack(Key)) || {Key, _} <- Rows].
+
+clear_index_range(Tx, ClearList, DbPrefix) ->
```

Comment:
these functions feel odd to me, since they don't run like functions (the clear doesn't happen until after the function returns).

Reply:
@rnewson are you saying that because the commit doesn't happen until the end of the transaction call, these functions may have been transient?

Reply:
what I mean is they don't do anything (yet). Maybe others won't find it as weird as I do. Let's focus on getting this right first and we can circle back on this quibble later.